src/components/LocalFileUpload/custom-request-upload.ts

```diff
@@ -1,39 +1,113 @@
-import { RcCustomRequestOptions } from "antd/lib/upload/interface";
+import { SimulariumFileFormat } from "@aics/simularium-viewer";
+import { findIndex } from "lodash";
+import {
+    UploadRequestOption,
+    UploadRequestError,
+} from "rc-upload/lib/interface";
+
 import { LocalSimFile } from "../../state/trajectory/types";
 import { CLEAR_SIMULARIUM_FILE } from "../../state/trajectory/constants";
 import { store } from "../..";
+
+// Typescript's File definition is missing this function
+//  which is part of the HTML standard on all browsers
+//  and needed below
+interface FileHTML extends File {
+    text(): Promise<string>;
+}
```

Comment:
This and much of this file from line 57 onward is copied (with some modifications) from the simularium-viewer testbed